### PR TITLE
Task #218156 [WEB] Learner Profile Update, Task #216421 [WEB] Common App Bar, Drawer & Footer Component, Task #216822 [WEB] Common Custom Card Component Modifications

### PR DESCRIPTION
**Pull request by Alfaj2302**
Learner Profile Update Task
https://tracker.tekdi.net/issues/218156

Drawer Task
https://tracker.tekdi.net/issues/216421

CardComponent Task
https://tracker.tekdi.net/issues/216822

### I have ensured that following `Pull Request Checklist` is taken care of before sending this PR
* [ ] Code is formatted as per format decided
* [ ] Updated acceptance criteria in tracker
* [ ] Updated test cases in test-cases-tracker

## Summary by CodeRabbit

- **New Features**
  - Implemented multilingual support with key translations in English and Hindi.
  - Introduced Checkbox component in BaseInput and revamped DashboardCard styling.
  - Enhanced state management and data operations in various dashboard components for a smoother user experience.
  - Optimized onboarding and profile management workflows with updated components and state handling.

- **Bug Fixes**
  - Adjusted text weight and removed specific style attributes for consistent UI presentation.

- **Style**
  - Updated color schemes and structural styling in components like FileUpload and DashboardCard for a cohesive visual design.

- **Documentation**
  - Enhanced translation documentation with new keys and modifications for improved clarity and extensibility.

### CODE REVIEW

**Review by rushi-tekdi — APPROVED** [review]

File: apps/front-end/src/v2/components/Functional/LearnerOnboardingDetails/LearnerDocsChecklist.js [hunk 1]
```diff
@@ -117,6 +143,244 @@ const LearnerDocsChecklist = ({ footerLinks }) => {
...
+            localStorage.setItem("loadCohort", "yes");
+            setIsUserRegisterExist(true);
+          } else {
+            setIsUserRegisterExist(false);
+            await showSelectCohort();
+          }
+        } catch (e) {}
+      }
+    }
+    fetchData();
```
[the remark this answers is not in the document]
Comment (Alfaj2302) [reply apps/front-end/src/v2/components/Functional/LearnerOnboardingDetails/LearnerDocsChecklist.js]: No need to Solve

File: apps/front-end/src/v2/components/Functional/LearnerOnboardingDetails/BenificiaryBasicDetails.js [hunk 2]
```diff
@@ -69,8 +96,256 @@ export default function BenificiaryBasicDetails() {
...
+        }
+        setLoading(false);
+        //end do page load first operation
+        setCountLoad(2);
+      } else if (countLoad == 2) {
+        setCountLoad(3);
+      }
+    }
+    fetchData();
+  }, [countLoad]);
```
[the remark this answers is not in the document]
Comment (Alfaj2302) [reply apps/front-end/src/v2/components/Functional/LearnerOnboardingDetails/BenificiaryBasicDetails.js:150]: No need to solve

File: apps/front-end/src/v2/components/Functional/LearnerOnboardingDetails/BenificiaryBasicDetails.js [hunk 3]
```diff
@@ -69,8 +96,256 @@ export default function BenificiaryBasicDetails() {
...
+      await Promise.all([
+        setIndexedDBItem("enums", ListOfEnum.data),
+        setIndexedDBItem("qualification", qualification),
+        setIndexedDBItem("lastFetchTime", currentTime),
+        setIndexedDBItem("editRequest", editRequest),
+      ]);
+    } catch (error) {
+      console.error("Error saving data to IndexedDB:", error);
+    }
+  };
```
[the remark this answers is not in the document]
Comment (Alfaj2302) [reply apps/front-end/src/v2/components/Functional/LearnerOnboardingDetails/BenificiaryBasicDetails.js:122]: Resolved

File: apps/front-end/src/v2/components/Functional/LearnerOnboardingDetails/BenificiaryBasicDetails.js [hunk 4]
```diff
@@ -9,18 +9,45 @@ import {
...
+  objProps,
+  arrList,
 } from "@shiksha/common-lib";
 import { useParams, useNavigate } from "react-router-dom";
 import moment from "moment";
 import ProfilePhoto from "../../Functional/ProfilePhoto/ProfilePhoto";
+import { getIpUserInfo, setIpUserInfo } from "v2/utils/SyncHelper/SyncHelper";
 
-export default function BenificiaryBasicDetails() {
+export default function BenificiaryBasicDetails(userTokenInfo) {
```
[the remark this answers is not in the document]
Comment (Alfaj2302) [reply apps/front-end/src/v2/components/Functional/LearnerOnboardingDetails/BenificiaryBasicDetails.js:25]: No need to solve

File: apps/front-end/src/pages/front-end/facilitator/FacilitatorQualification.js [hunk 5]
```diff
@@ -11,11 +11,36 @@ import {
...
+            key={index}
+            t={t}
+            enumOptionValue={item}
+            {...{ enumType, enumApiData }}
+          />
+        </Text>
+      ))}
+    </VStack>
+  );
+};
```
[the remark this answers is not in the document]
Comment (Alfaj2302) [reply apps/front-end/src/pages/front-end/facilitator/FacilitatorQualification.js:42]: No need to solve

File: apps/front-end/src/pages/front-end/facilitator/FacilitatorBasicDetails.js [hunk 6]
```diff
@@ -82,12 +108,254 @@ export default function FacilitatorBasicDetails({ userTokenInfo }) {
...
+      if (user_cohort_list?.data.length == 1) {
+        setSelectCohortForm(false);
+        await checkDataToIndex();
+        await checkUserToIndex();
+      } else {
+        setSelectCohortForm(true);
+      }
+    }
+  };
+  const checkDataToIndex = async () => {
```
[the remark this answers is not in the document]
Comment (Alfaj2302) [reply apps/front-end/src/pages/front-end/facilitator/FacilitatorBasicDetails.js:320]: No need to solve

File: apps/front-end/src/pages/front-end/facilitator/FacilitatorBasicDetails.js [hunk 7]
```diff
@@ -82,12 +108,254 @@ export default function FacilitatorBasicDetails({ userTokenInfo }) {
...
+        const user_cohort_id = academicYear;
+        const cohort_data = await facilitatorRegistryService.getCohort({
+          cohortId: user_cohort_id,
+        });
+        setSelectedCohortData(cohort_data);
+        await setSelectedAcademicYear(cohort_data);
+      }
+    }
+    fetchData();
+  }, [academicYear]);
```
[the remark this answers is not in the document]
Comment (Alfaj2302) [reply apps/front-end/src/pages/front-end/facilitator/FacilitatorBasicDetails.js:224]: No need to solve

File: apps/front-end/src/pages/front-end/facilitator/FacilitatorBasicDetails.js [hunk 8]
```diff
@@ -82,12 +108,254 @@ export default function FacilitatorBasicDetails({ userTokenInfo }) {
...
+        }
+        setLoading(false);
+        //end do page load first operation
+        setCountLoad(2);
+      } else if (countLoad == 2) {
+        setCountLoad(3);
+      }
+    }
+    fetchData();
+  }, [countLoad]);
```
[the remark this answers is not in the document]
Comment (Alfaj2302) [reply apps/front-end/src/pages/front-end/facilitator/FacilitatorBasicDetails.js:161]: No Need to solve

File: apps/front-end/src/pages/front-end/facilitator/FacilitatorBasicDetails.js [hunk 9]
```diff
@@ -82,12 +108,254 @@ export default function FacilitatorBasicDetails({ userTokenInfo }) {
...
+      await Promise.all([
+        setIndexedDBItem("enums", ListOfEnum.data),
+        setIndexedDBItem("qualification", qualification),
+        setIndexedDBItem("lastFetchTime", currentTime),
+        setIndexedDBItem("editRequest", editRequest),
+      ]);
+    } catch (error) {
+      console.error("Error saving data to IndexedDB:", error);
+    }
+  };
```
[the remark this answers is not in the document]
Comment (Alfaj2302) [reply apps/front-end/src/pages/front-end/facilitator/FacilitatorBasicDetails.js:133]: No need to solve

File: apps/front-end/src/v2/components/Functional/LearnerOnboardingDetails/LearnerDocsChecklist.js [hunk 10]
```diff
@@ -117,6 +143,244 @@ const LearnerDocsChecklist = ({ footerLinks }) => {
...
+      await Promise.all([
+        setIndexedDBItem("enums", ListOfEnum.data),
+        setIndexedDBItem("qualification", qualification),
+        setIndexedDBItem("lastFetchTime", currentTime),
+        setIndexedDBItem("editRequest", editRequest),
+      ]);
+    } catch (error) {
+      console.error("Error saving data to IndexedDB:", error);
+    }
+  };
```
[the remark this answers is not in the document]
Comment (Alfaj2302) [reply apps/front-end/src/v2/components/Functional/LearnerOnboardingDetails/LearnerDocsChecklist.js]: Resolved

File: apps/front-end/src/v2/components/Static/FormBaseInput/FormBaseInput.js [hunk 11]
```diff
@@ -288,7 +288,7 @@ export const FieldTemplate = ({
   return (
     <VStack
       style={style}
-      space={id === "root" && label ? "10" : schema?.label ? "4" : "0"}
+      space={id === "root" && label ? "6" : schema?.label ? "4" : "0"}
```
[the remark this answers is not in the document]
Comment (Alfaj2302) [reply apps/front-end/src/v2/components/Static/FormBaseInput/FormBaseInput.js:291]: Resolved

File: apps/front-end/src/v2/components/Functional/LearnerOnboardingDetails/LearnerDocsChecklist.js [hunk 12]
```diff
@@ -117,6 +143,244 @@ const LearnerDocsChecklist = ({ footerLinks }) => {
...
+      await Promise.all([
+        setIndexedDBItem("enums", ListOfEnum.data),
+        setIndexedDBItem("qualification", qualification),
+        setIndexedDBItem("lastFetchTime", currentTime),
+        setIndexedDBItem("editRequest", editRequest),
+      ]);
+    } catch (error) {
+      console.error("Error in Promise.all fetching data:", error);
+    }
+  };
```
[the remark this answers is not in the document]
Comment (Alfaj2302) [reply apps/front-end/src/v2/components/Functional/LearnerOnboardingDetails/LearnerDocsChecklist.js]: Resolved

**Pull request merged**